lib/galaxy/webapps/galaxy/api/cloud.py

```diff
@@ -3,255 +3,101 @@
 """
 
 import logging
+from typing import Union
 
-from galaxy import exceptions
-from galaxy.exceptions import ActionInputError
-from galaxy.managers import (
-    cloud,
-    datasets,
+from fastapi import (
+    Body,
+    Response,
+    status,
 )
-from galaxy.structured_app import StructuredApp
-from galaxy.web import expose_api
-from . import BaseGalaxyAPIController
+from pydantic import Required
+
+from galaxy.managers.cloud import CloudManager
+from galaxy.managers.context import ProvidesHistoryContext
+from galaxy.managers.datasets import DatasetSerializer
+from galaxy.schema.cloud import (
+    CloudDatasets,
+    CloudObjects,
+    InputArguments,
+    StatusCode,
+    SummaryGetObjects,
+    SummarySendDatasets,
+)
+from galaxy.webapps.galaxy.api import (
+    depends,
+    Router,
+)
+from . import DependsOnTrans
 
 log = logging.getLogger(__name__)
 
+router = Router(tags=["cloud"])
 
-class CloudController(BaseGalaxyAPIController):
-    """
-    RESTfull controller for interaction with Amazon S3.
-    """
-
-    def __init__(
-        self, app: StructuredApp, cloud_manager: cloud.CloudManager, datasets_serializer: datasets.DatasetSerializer
-    ):
-        super().__init__(app)
-        self.cloud_manager = cloud_manager
-        self.datasets_serializer = datasets_serializer
-
-    @expose_api
-    def index(self, trans, **kwargs):
-        """
-        GET /api/cloud/storage
 
-        Lists cloud-based buckets (e.g., S3 bucket, Azure blob) user has defined.
+@router.cbv
+class FastAPICloudController:
```

Comment:
Can you mark the new routes as deprecated (deprecated=True) in the route definition ?

Reply:
It is done.